ompi/mca/pml/ucx/pml_ucx.c

```diff
@@ -368,74 +368,32 @@ static inline ucp_ep_h mca_pml_ucx_get_ep(ompi_communicator_t *comm, int rank)
     return NULL;
 }
 
-static void mca_pml_ucx_waitall(void **reqs, int *count_p)
-{
-    int i;
-
-    PML_UCX_VERBOSE(2, "waiting for %d disconnect requests", *count_p);
-    for (i = 0; i < *count_p; ++i) {
-        opal_common_ucx_wait_request(reqs[i], ompi_pml_ucx.ucp_worker, "ucp_disconnect_nb");
-        reqs[i] = NULL;
-    }
-
-    *count_p = 0;
-}
-
 int mca_pml_ucx_del_procs(struct ompi_proc_t **procs, size_t nprocs)
 {
     ompi_proc_t *proc;
-    int num_reqs;
-    size_t max_reqs;
-    void *dreq, **dreqs;
-    ucp_ep_h ep;
+    opal_common_ucx_del_proc_t *del_procs;
     size_t i;
+    int ret;
 
-    max_reqs = ompi_pml_ucx.num_disconnect;
-    if (max_reqs > nprocs) {
-        max_reqs = nprocs;
-    }
-
-    dreqs = malloc(sizeof(*dreqs) * max_reqs);
-    if (dreqs == NULL) {
+    del_procs = malloc(sizeof(*del_procs) * nprocs);
+    if (del_procs == NULL) {
         return OMPI_ERR_OUT_OF_RESOURCE;
     }
 
-    num_reqs = 0;
-
     for (i = 0; i < nprocs; ++i) {
         proc = procs[(i + OMPI_PROC_MY_NAME->vpid) % nprocs];
-        ep   = proc->proc_endpoints[OMPI_PROC_ENDPOINT_TAG_PML];
-        if (ep == NULL) {
-            continue;
-        }
+        del_procs[i].ep   = proc->proc_endpoints[OMPI_PROC_ENDPOINT_TAG_PML];
```

Comment:
let's move the `i + OMPI_PROC_MY_NAME->vpid) % nprocs` part to common code

Reply:
will not work because spml uses own procs list array.
actually % nprocs is not required at all - we always iterate all procs and disconnect it

Reply:
we can pass OMPI_PROC_MY_NAME->vpid as "int my_index" parameter to opal_common_ucx_del_procs(), and this function can do "proc=procs[(my_index + i) % count]"
need to do it to avoid many-to-one scenario in disconnect flow

Reply:
got it. fixed